main/cache/BUILD

```diff
@@ -0,0 +1,19 @@
+cc_library(
+    name = "cache",
```

Comment:
Cache is a dependency of realmain and realmain-orig.

I think this dependency will now pull in the full //main:pipeline for realmain-orig.

Is there a way to avoid that? Do I need to create a cache-orig too?

Reply:
> Is there a way to avoid that? Do I need to create a cache-orig too?

that'd be nice. Just a version that always returns nullptr.

This will also go well with emscripten, as we don't want to bring in a database into it

Reply:
Ah, so `realmain-orig` doesn't need caching? Excellent. I was already planning on making a null version for Emscripten, as it seemed beneficial when we move the caching logic out of `pipeline`.